sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -829,6 +917,68 @@ private void processElementForWindowObservingSplitRestriction(
     this.stateAccessor.finalizeState();
   }
 
+  private void processElementForTruncateSizedRestriction(
+      WindowedValue<KV<KV<InputT, KV<RestrictionT, WatermarkEstimatorStateT>>, Double>> elem) {
+    currentElement = elem.withValue(elem.getValue().getKey().getKey());
+    currentRestriction = elem.getValue().getKey().getValue().getKey();
+    currentWatermarkEstimatorState = elem.getValue().getKey().getValue().getValue();
+    currentTracker =
```

Comment:
It looks like currentTracker isn't being set within both `processElementForSplitRestriction` methods. Could you fix that in this PR?

Reply:
Yeah I'm going to update it.